src/MCPClient/lib/clientScripts/archivematicaUpdateSizeAndChecksum.py

```diff
@@ -17,26 +17,127 @@
-# @author Joseph Perry <joseph@artefactual.com>
-from optparse import OptionParser
+import argparse
+import os
+import re
+import sys
 
 # fileOperations requires Django to be set up
 import django
 django.setup()
-# archivematicaCommon
+
+from main.models import File, Transfer
+
 from custom_handlers import get_script_logger
 from fileOperations import updateSizeAndChecksum
+from option_parsers import UUIDArgAction
+
+import metsrw
+
+
+logger = get_script_logger('archivematica.mcp.client.updateSizeAndChecksum')
+
+
+def find_mets_file(unit_path):
+    """
+    Return the location of the original METS in a Archivematica AIP transfer.
+    """
+    p = re.compile(r'^METS\..*\.xml$', re.IGNORECASE)
+    src = os.path.join(unit_path, 'metadata')
+    for item in os.listdir(src):
+        m = p.match(item)
+        if m:
+            return os.path.join(src, m.group())
+
+
+def get_size_and_checksum_from_mets(shared_path, file_):
+    """
+    Given an instance of a File, return a tuple with three values: file_Size,
+    checksum and checksum_type, as they are described in the original METS
+    document of the transfer. The values will be None when not found.
+    """
+    transfer = file_.transfer
+    transfer_location = transfer.currentlocation.replace('%sharedPath%', shared_path, 1)
+
+    mets_file = find_mets_file(transfer_location)
+    if not mets_file:
+        logger.info('Archivematica AIP: METS file not found in %s.', transfer_location)
+        return
+
+    logger.info('Archivematica AIP: reading METS file %s.', mets_file)
+    mets = metsrw.METSDocument.fromfile(mets_file)
+
+    fsentry = mets.get_file(file_.uuid)
+    if not fsentry:
+        logger.error('Archivematica AIP: FSEntry with UUID %s not found', file_.uuid)
+        return None
+    if not len(fsentry.amdsecs):
+        logger.error('Archivematica AIP: FSEntry.amdsecs is empty')
+        return None
+    amdsec = fsentry.amdsecs[0]
+    for item in amdsec.subsections:
+        if item.subsection == 'techMD':
+            techmd = item
+    if not techmd:
+        logger.error('Archivematica AIP: techMD section could not be found')
+        return None
+    if techmd.contents.mdtype != 'PREMIS:OBJECT':
+        logger.error('Archivematica AIP: PREMIS:OBJECT could not be found')
+        return None
+    pobject = techmd.contents.document # Element
+
+    size = checksum = checksum_type = None
+    r = pobject.xpath('premis:objectCharacteristics/premis:objectCharacteristicsExtension/Mediainfo/File/track[@type="General"]/File_size', namespaces=metsrw.utils.NAMESPACES)
+    if r:
+        size = r[0].text
+    r = pobject.xpath('premis:objectCharacteristics/premis:fixity/premis:messageDigest', namespaces=metsrw.utils.NAMESPACES)
+    if r:
+        checksum = r[0].text
+    r = pobject.xpath('premis:objectCharacteristics/premis:fixity/premis:messageDigestAlgorithm', namespaces=metsrw.utils.NAMESPACES)
+    if r:
+        checksum_type = r[0].text
```

Comment:
After later discussion, we decided not to generate checksums, but just to use the ones from the METS file, so this is good.